client/extensions/woocommerce/state/ui/products/selectors.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { get, find, isNumber } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getProduct } from '../../products/selectors';
+
+function getEditsState( state ) {
+	const woocommerce = state.extensions.woocommerce;
+	return get( woocommerce, 'ui.products.edits' );
+}
+
+/**
+ * Gets the accumulated edits for a product, if any.
+ *
+ * @param {Object} state Global state tree
+ * @param {any} productId The id of the product (or { index: # } )
+ * @return {Object} The current accumulated edits
+ */
+export function getProductEdits( state, productId ) {
+	const edits = getEditsState( state ) || {};
+	const bucket = isNumber( productId ) && 'updates' || 'creates';
+	const array = edits[ bucket ] || [];
+
+	return find( array, ( p ) => productId === p.id );
+}
+
+/**
+ * Gets a combined product ( fetched data + edits )
+ *
+ * @param {Object} state Global state tree
+ * @param {any} productId The id of the product (or { index: # } )
+ * @return {Object} The product data merged between the fetched data and edits
+ */
+export function getCombinedProduct( state, productId ) {
```

Comment:
Would you consider `getProductWithLocalEdits` instead to be more clear down the road?